parsing_input.c

```diff
@@ -0,0 +1,61 @@
+#include "shell.h"
+/**
+ * calculate_tokens - function to calculate token command
+ * @lineptr: pointer to string
+ * @delim: delimiter between strings
+ * Return: number of tokens
+ */
+int calculate_tokens(char *lineptr, char *delim)
+{
+	int counter = 0;
+	char *token, *line_copy;
+
+	line_copy = _strcpy(line_copy, lineptr);
+	token = strtok(line_copy, delim);
+	while (token != NULL)
+	{
+		counter++;
+		token = strtok(line_copy, delim);
```

Comment:
calling strtok function a second time, the you pass NULL as the string else, it wont move to the next token.. Check your code